libraries/EEPROM/EEPROM.h

```diff
@@ -53,8 +53,11 @@ class EEPROMClass {
     if (address < 0 || address + sizeof(T) > _size)
       return t;
 
-    memcpy(_data + address, (const uint8_t*) &t, sizeof(T));
-    _dirty = true;
+    _dirty = (memcmp(_data + address, (const uint8_t*) &t, sizeof(T)) !=0);
```

Comment:
Is it really OK to overwrite _dirty? Consider this scenario:
1. EEPROM.write is called. _dirty is set to true
2. EEPROM.put is called with a struct that is identical to what is already on the flash. _dirty is set to false
3. EEPROM.commit is called. Since _dirty is false, commit returns true without actually writing anything.

I'm not an expert c/c++ programmer, so I might have missed something. Sorry for bothering you if that is the case.